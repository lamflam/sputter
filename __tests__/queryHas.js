const snapshot = require("../testUtil/snapshot");
const fixtures = require("../testUtil/fixtures");

describe("Parent selector query", () => {
  describe("conditional", () => {
    snapshot(
      fixtures.conditional,
      'ExpressionStatement:has([name="foo"][type="Identifier"])'
    );
  });

  describe("one of", () => {
    snapshot(
      fixtures.conditional,
      'IfStatement:has(LogicalExpression [name="foo"], LogicalExpression [name="x"])'
    );
  });

  describe("chaining", () => {
    snapshot(
      fixtures.conditional,
      'BinaryExpression:has(Identifier[name="x"]):has(Literal[value="test"])'
    );
  });

  describe("nesting", () => {
    snapshot(
      fixtures.conditional,
      "Program:has(IfStatement:has(Literal[value=true], Literal[value=false]))"
    );
  });

  describe("non-matching", () => {
    snapshot(fixtures.conditional, ':has([value="impossible"])');
  });
});
