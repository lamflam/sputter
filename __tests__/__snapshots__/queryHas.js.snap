// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Has query chaining BinaryExpression:has(Identifier[name="x"]):has(Literal[value="test"]) 1`] = ``;

exports[`Has query conditional ExpressionStatement:has([name="foo"][type="Identifier"]) 1`] = `
  1 | if (x === 1) {
> 2 |   foo();
    |   ^^^^^^ ExpressionStatement
  3 | } else {
  4 |   x = 2;
  5 | }
`;

exports[`Has query nesting Program:has(IfStatement:has(Literal[value=true], Literal[value=false])) 1`] = ``;

exports[`Has query non-matching :has([value="impossible"]) 1`] = ``;

exports[`Has query one of IfStatement:has(LogicalExpression [name="foo"], LogicalExpression [name="x"]) 1`] = `
   4 |   x = 2;
   5 | }
>  6 | if ((x == "test" && true) || x) {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>  7 |   y = -1;
     | ^^^^^^^^^
>  8 | } else if (false) {
     | ^^^^^^^^^
>  9 |   y = 1;
     | ^^^^^^^^^
> 10 | }
     | ^^ IfStatement
  11 | 
`;
